http.go

```diff
@@ -5,11 +5,52 @@ import (
 	"fmt"
 	"net"
 	"net/http"
-	"net/url"
 	"runtime/debug"
 	"strings"
 )
 
+const (
+	redaction = "********"
+)
+
+// Query fields whose value will redacted. Used by NewHttp.
+var QuerySecretFields = []string{"password", "passphrase", "passwd", "secret"}
+
+// Header fields whose value will redacted. Used by NewHttp.
+var HeaderSecretFields = []string{"Authorization"}
+
+func redactQuery(r *http.Request) string {
+	query := r.URL.Query()
+
+	for _, keyword := range QuerySecretFields {
+		for field := range query {
+			if field == keyword {
+				query[field] = []string{redaction}
+				break
+			}
+		}
+	}
+
+	return query.Encode()
+}
+
+func redactHeaders(r *http.Request) map[string]string {
+	headers := make(map[string]string, len(r.Header))
+
+	for k, v := range r.Header {
+		for _, field := range HeaderSecretFields {
+			if field == k {
+				rep := strings.Repeat(redaction+",", len(v))
```

Comment:
Why are we turning the list of headers `[]string` into a single string? We should preserve this imo.

What about doing something like this:

```go
func redactHeaders(r *http.Request) map[string]string {
    headers := make(map[string][]string, len(r.Header))
    for k, v := range r.Header {
        headers[k] = make([]string, len(v))
        for _, field := range HeaderSecretFields {
            if field == k {
                for i, _ := range v {
                    headers[k][i] = redaction
                }
            } else {
                copy(headers[k], v)
            }
        }
    }
    return headers
}
```